src/lib/tracking/trackProductListViewed.js

```diff
@@ -1,41 +1,18 @@
-/* eslint-disable camelcase */
-import React from "react";
 import track from "./track";
+import getProductListTrackingData from "./utils/getProductListTrackingData";
 
 /**
  * trackProductListViewed higher tracks a list of viewed products
  * @name trackProductListViewed
  * @param {Object} options options to supply to tracking HOC
  * @returns {React.Component} - component
  */
-export default (options) => (Component) => {
-  const component = class trackProductListViewed extends React.Component {
-    render() {
-      return (
-        <Component {...this.props} />
-      );
-    }
+export default (options) => track(({ tag, catalogItems }) => {
+  const products = (Array.isArray(catalogItems) && catalogItems.map((catalogItem) => catalogItem.node.product)) || undefined;
```

Comment:
Do we need a protection here to check for `node`, similar to the other Segment PR? We might not, I haven't had any errors yet.

Reply:
I don't think so, because node should exist if there are items in the array.